integrations/pagerduty/main.py

```diff
@@ -0,0 +1,49 @@
+from typing import Any, Dict, List
+from clients.pagerduty import PagerDutyClient
+from clients.utils import ObjectKind
+from port_ocean.context.ocean import ocean
+from starlette.requests import Request
+
+pager_duty_client = PagerDutyClient(
+    ocean.integration_config["token"],
+    ocean.integration_config["url"],
+    ocean.integration_config["app_url"],
+)
+
+
+@ocean.on_resync(ObjectKind.INCIDENTS)
+async def on_incidents_resync(kind: str) -> List[Dict[Any, Any]]:
+    incidents = pager_duty_client.paginate_request_to_pager_duty(data_key="incidents")
+    return incidents
+
+
+@ocean.on_resync(ObjectKind.SERVICES)
+async def on_services_resync(kind: str) -> List[Dict[Any, Any]]:
+    services = pager_duty_client.paginate_request_to_pager_duty(data_key="services")
+    return services
+
+
+@ocean.router.post("/webhook")
+async def upsert_incident_webhook_handler(
+    data: Dict[str, Any], request: Request
+) -> None:
+    if data["event"]["event_type"] in pager_duty_client.service_delete_events:
```

Comment:
Make sure the webhook body is identical to the api
We dont want to have different jq's for the webhooks and the syncs